### PR TITLE
Decompile unk_020648EC

### PR DESCRIPTION
**Pull request by abaresk**

### CODE REVIEW

File: src/unk_020648EC.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+typedef void (*FieldSystemFunc1)(FieldSystem*);
+typedef BOOL (*FieldSystemFunc2)(FieldSystem*, u32, u32, u32, u32);
+
+static const FieldSystemFunc1 _020FE214[];
+static const FieldSystemFunc1 _020FE1EC[];
+static const FieldSystemFunc2 _020FE23C[];
+
+void sub_020648EC(FieldSystem* fsys) {
+    int gymmickType = SavGymmick_GetType(Sav2_GetGymmickPtr(Fsys_GetSaveDataPtr(fsys)));
+    if (gymmickType) {
```
Comment (PikalaxALT) [inline src/unk_020648EC.c:16]: `gymmickType` is not a boolean, so explicit comparison to `GYMMICK_NONE` is desired for the sake of communication.

File: src/unk_020648EC.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        _020FE214[gymmickType](fsys);
+    }
+}
+
+void sub_02064910(FieldSystem* fsys) {
+    int gymmickType = SavGymmick_GetType(Sav2_GetGymmickPtr(Fsys_GetSaveDataPtr(fsys)));
+    if (gymmickType) {
+        FieldSystemFunc1 func = _020FE1EC[gymmickType];
+        if (func) {
```
Comment (PikalaxALT) [inline src/unk_020648EC.c:25]: `func` is not a boolean, so explicit comparison to `NULL` is desired for the sake of communication.

File: src/unk_020648EC.c [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            func(fsys);
+        }
+    }
+}
+
+BOOL sub_02064938(FieldSystem *fsys, u32 a1, u32 a2, u32 a3, u32 a4) {
+    int gymmickType = SavGymmick_GetType(Sav2_GetGymmickPtr(Fsys_GetSaveDataPtr(fsys)));
+    if (!gymmickType) return FALSE;
```
Comment (PikalaxALT) [inline src/unk_020648EC.c:33]:
```suggestion
    if (gymmickType == GYMMICK_NONE) {
        return FALSE;
    }
```

File: src/unk_020648EC.c [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    FieldSystemFunc2 func = _020FE23C[gymmickType];
+    if (!func) return FALSE;
```
Comment (PikalaxALT) [inline src/unk_020648EC.c:36]:
```suggestion
    if (func == NULL) {
        return FALSE;
    }
```

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: src/unk_020648EC.c (hunk 1 above)
Comment (abaresk) [reply src/unk_020648EC.c]: Done

File: src/unk_020648EC.c (hunk 2 above)
Comment (abaresk) [reply src/unk_020648EC.c]: Done

File: src/unk_020648EC.c (hunk 4 above)
Comment (abaresk) [reply src/unk_020648EC.c]: Done

File: src/unk_020648EC.c (hunk 3 above)
Comment (abaresk) [reply src/unk_020648EC.c]: Done

**Review by ScottNortonPhD — APPROVED** [review]

**Review by PikalaxALT — APPROVED** [review]

**Pull request merged**